addons/mrp/models/mrp_unbuild.py

```diff
@@ -132,6 +132,28 @@ def _unlink_except_done(self):
         if 'done' in self.mapped('state'):
             raise UserError(_("You cannot delete an unbuild order if the state is 'Done'."))
 
+    def _create_finished_move_line(self, finished_move):
+        return self.env['stock.move.line'].create({
```

Comment:
Both of these methods create a new object. Is there a reason why they also return it? It is not used in the current code, do you expect this will be helpful when extending it ?

Reply:
I returned these created new objects because it want to assign the new fields by extending it(eg. owner_id).
With odoo current design, it is hard coded and we can't extend it. This is how I want to use from custom module.
```
def _create_move_line(self, move, origin_move_line, taken_quantity):
        move_line = super(MrpUnbuild, self)._create_move_line(
            move, origin_move_line, taken_quantity
        )
        move_line.write({"owner_id": origin_move_line.owner_id})
        return move_line
```

Reply:
Do you plan to extend the object only by adding new/modifying existing values, or do you also expect that some methods might be called on the objects? If it's the former, than perhaps we could return a dictionary of values from the newly extracted methods, and then create objects in the original place, based on those values?

Reply:
Thanks for the suggestion. I updated the code.